syncer/syncer.go

```diff
@@ -641,66 +644,60 @@ func (s *Syncer) getMasterStatus() (mysql.Position, gtid.Set, error) {
 	return s.fromDB.getMasterStatus(s.cfg.Flavor)
 }
 
-// clearTables is used for clear table cache of given table. this function must
-// be called when DDL is applied to this table.
-func (s *Syncer) clearTables(schema, table string) {
-	key := dbutil.TableName(schema, table)
-	delete(s.tables, key)
-	delete(s.cacheColumns, key)
-	s.genColsCache.clearTable(schema, table)
-}
-
-func (s *Syncer) clearAllTables() {
-	s.tables = make(map[string]*table)
-	s.cacheColumns = make(map[string][]string)
-	s.genColsCache.reset()
-}
-
-func (s *Syncer) getTableFromDB(db *DBConn, schema string, name string) (*table, error) {
-	table := &table{}
-	table.schema = schema
-	table.name = name
-	table.indexColumns = make(map[string][]*column)
+func (s *Syncer) getTable(origSchema, origTable, renamedSchema, renamedTable string, p *parser.Parser) (*model.TableInfo, error) {
+	ti, err := s.schemaTracker.GetTable(origSchema, origTable)
+	if err == nil || !schema.IsTableNotExists(err) {
```

Comment:
will ti be nil if err is not nil?

Reply:
`(*schema.Tracker).GetTable()` will either return `ti, nil` or `nil, err` (Unless a TiDB table can have no TableInfo).